spacy_llm/tasks/span/examples.py

```diff
@@ -1,13 +1,17 @@
 import abc
-from typing import Dict, Iterable, List
+from typing import Dict, Generic, Iterable, List
 
 from spacy.tokens import Span
 
-from ...compat import BaseModel
-from ...ty import FewshotExample, Self
+from ...compat import BaseModel, Self
+from ...ty import FewshotExample, TaskContraT
 
 
-class SpanExample(FewshotExample, abc.ABC):
+class SpanExample(FewshotExample[TaskContraT], abc.ABC, Generic[TaskContraT]):
+    """Example for span tasks not using CoT.
+    Note: this should be SpanTaskContraT instead of TaskContraT, but this would entail a circular import.
```

Comment:
Isn't there any other solution? A circular import typically means we should reorganize what is in which module...

Reply:
I agree, but in this case: 
- `SpanExample` should use `FewshotExample` parametrized with `SpanTaskContraT`: `FewshotExample[SpanTaskContraT]`.
- `SpanTaskContraT` requires `SpanTask` in its definition.
- `SpanTask` requires `SpanExample`.

This is a dependency loop, i. e. doing this in multiple files will (without local imports) always lead to circular import errors. The one alternative here would be to drop everything in `task.py` - also not nice :man_shrugging: